### DISCUSSION TITLE
Generate ziggy.js file from specific route.php file

### DISCUSSION THREAD

**Discussion by aleksandar-atanasov** [tighten/ziggy#503] [Q&A] [answered]
Is there an option to pass an additional parameter to php artisan ziggy:generate and pass specific php routes file like admin.php and generate only the routes from that file not all the routes files inside the routes directory?

**Comment by bakerkretzmar** [comment, marked as answer]
Yeah, configure a [route group](https://github.com/tighten/ziggy#filtering-using-groups) and then pass it to the command with the `--group` option. You can't point it to a specific file but if all those route names start with `admin.` or something it'll be very simple to set up.

**Reply by aleksandar-atanasov** [reply]
When trying to access route helper inside the vue script it says `Uncaught (in promise) ReferenceError: route is not defined`
but when accessed in the vue template it works just as expected.
I'm using the vue script setup
I removed the @routes from the blade template
This is my app.js setup

```import { createApp, h } from 'vue'
import { createInertiaApp } from '@inertiajs/inertia-vue3'
import { ZiggyVue } from 'ziggy';
import { Ziggy } from './ziggy';

createInertiaApp({
  resolve: name => require(`./Pages/${name}`),
  setup({ el, App, props, plugin }) {
    createApp({ render: () => h(App, props) })
      .use(plugin)
      .use(ZiggyVue,Ziggy)
      .mount(el)
  },
});```

**Reply by bakerkretzmar** [reply]
With `<script setup>` you'll have to import the `route` function every time you need it or use `provide` and `inject`. See this discussion for a few examples: https://github.com/tighten/ziggy/discussions/492
